### PR TITLE
fix: [MUSD-857] show native amount for erc20 money account deposits

### PR DESCRIPTION
**Pull request by Jwhiles**
## **Description**
This fixes a bug where we would miscalculate the token value shown for ERC20 token deposits to the money account. The root of the problem was that we calculated the token values assuming an incorrect number of decimal places.

 

## **Changelog**

CHANGELOG entry: show correct token values for ERC20 token deposits to money account

## **Related issues**

Fixes: [musd-857](https://consensyssoftware.atlassian.net/browse/MUSD-857?atlOrigin=eyJpIjoiZWIyNmU2Y2U2Y2U2NGNmODk3ZjA2NDJhNTFkNGE4YzUiLCJwIjoiaiJ9)

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

### **Before**
image

### **After**
image

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I've included tests if applicable
- [ ] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

#### Performance checks (if applicable)

- [ ] I've tested on Android
  - Ideally on a mid-range device; emulator is acceptable
- [ ] I've tested with a power user scenario
  - Use these [power-user SRPs](https://consensyssoftware.atlassian.net/wiki/spaces/TL1/pages/edit-v2/401401446401?draftShareId=9d77e1e1-4bdc-4be1-9ebb-ccd916988d93) to import wallets with many accounts and tokens
- [ ] I've instrumented key operations with Sentry traces for production performance metrics
  - See [`trace()`](/app/util/trace.ts) for usage and [`addToken`](/app/components/Views/AddAsset/components/AddCustomToken/AddCustomToken.tsx#L274) for an example

For performance guidelines and tooling, see the [Performance Guide](https://consensyssoftware.atlassian.net/wiki/spaces/TL1/pages/400085549067/Performance+Guide+for+Engineers).

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Changes user-visible transaction amounts for Money/Pay deposits and depends on live rates; scope is limited to display logic in the Money activity hook with strong test coverage.
> 
> **Overview**
> Fixes **MetaMask Pay money-account deposit** activity rows where the **pay-token primary line** was wrong for ERC-20s (e.g. misleading **+0.00 LINK** for MUSD-857).
> 
> **`useMoneyTransactionDisplayInfo`** now treats `requiredAssets[0].amount` as the **6-decimal USD deposit value** for every pay token, then divides by that token’s **USD price**—the same model already used for native ETH. ERC-20 USD price comes from **token→ETH market data × ETH→USD** via a shared **`getPayTokenUsdPrice`** helper; **`getTokenToEthPrice`** is exported from **`moneyActivityFiat`** for that lookup.
> 
> **Stablecoins** (USDC, USDT, etc.) are **priced from the API**, not assumed $1; the stable symbol list only drives **2-decimal display** vs up to **6 trimmed decimals** for other tokens. The primary line is **left blank** when rates are missing or the computed amount would show as zero, while the fiat line still uses existing logic.
> 
> Tests are expanded for **USDC** (market price, de-peg, no rates), **LINK** regressions, and related edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bc520652cd6fcf425342664d62abcdc53714ba1b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: app/components/UI/Money/hooks/useMoneyTransactionDisplayInfo.ts [hunk 1]
```diff
@@ -180,25 +186,112 @@ function getRequiredAsset(tx: TransactionMeta): RequiredAsset | undefined {
...
-  const humanReadable = fromTokenMinimalUnit(rawAmount, decimals);
-  const num = parseFloat(humanReadable);
-  if (isNaN(num)) {
+  if (isStable) {
+    const formatted = getIntlNumberFormatter(I18n.locale, {
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+      useGrouping: true,
+    }).format(amount.toNumber());
+    return `+${formatted} ${symbol}`;
```
[the remark this answers is not in the document]
Comment (Matt561) [reply app/components/UI/Money/hooks/useMoneyTransactionDisplayInfo.ts:287]: > A positive pay-token amount below half a cent (e.g. ~$0.004 USDC) can still render as +0.00 USDC,

@cursoragent this seems very unlikely to happen. Additionally, the lowest denomination of USD is a cent so we likely wouldn't want to display sub-cent values. Ultimately a product decision. I think you're overestimating the severity here.

**Review by Matt561 — APPROVED** [review]
